From d729564546bacbb0356ed249378a246229e854ed Mon Sep 17 00:00:00 2001
From: Leo Yan <leo.yan@intel.com>
Date: Wed, 11 Apr 2012 12:45:44 +0800
Subject: [PATCH] mfld_gi/ALS: Multiply the ALS raw data by glass light transmittance

BZ: 30035

We need to multiply ALS device raw data by the glass light
transmittance value to get the right lux value.

Change-Id: Ibf67812f340b66e90b727c53d6ee69622dc38fed
Signed-off-by: Leo Yan <leo.yan@intel.com>
---
 ctp_pr0/LightSensor.cpp |    3 ++-
 ctp_pr0/LightSensor.h   |    1 +
 2 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/ctp_pr0/LightSensor.cpp b/ctp_pr0/LightSensor.cpp
index efb80c9..fd11c30 100644
--- a/ctp_pr0/LightSensor.cpp
+++ b/ctp_pr0/LightSensor.cpp
@@ -77,7 +77,8 @@ int LightSensor::readEvents(sensors_event_t* data, int count)
     *data = mPendingEvent;
     data->timestamp = getTimestamp();
     read(data_fd, &val, sizeof(unsigned int));
-    data->light = (float)val / APDS9900_LUX_OUTPUT_SCALE;
+    data->light = (float)val /
+                    (APDS9900_LUX_OUTPUT_SCALE * APDS9900_LIGHT_TRANS);
     D("LightSensor - read data val = %f ",data->light);
 
     return 1;
diff --git a/ctp_pr0/LightSensor.h b/ctp_pr0/LightSensor.h
index f4495f6..4443e3e 100644
--- a/ctp_pr0/LightSensor.h
+++ b/ctp_pr0/LightSensor.h
@@ -27,6 +27,7 @@
 #include "InputEventReader.h"
 
 #define APDS9900_LUX_OUTPUT_SCALE 10.0
+#define APDS9900_LIGHT_TRANS 0.3
 
 /*****************************************************************************/
 
-- 
1.7.1

